From b84ef5f2f22b403818f33866c9a503b7acb4a39c Mon Sep 17 00:00:00 2001
From: Brinly Taylor <uberlaggydarwin@gmail.com>
Date: Sat, 7 Feb 2015 15:29:00 +1030
Subject: [PATCH 28/51] FM: Add property to force use internal antenna

Allow devices to force the FM radio to use the internal antenna setup by
setting hw.fm.internal_antenna property to true in system.prop.

Device tree example here: I7c9a67371c775a343345b035c196152bd9b2e13d

Change-Id: I1a055070bb4b7e49b7a03c75632d4b95c59f02da

Author: Dhruv Paranjape <lord.dhruv@gmail.com>
Date:   Thu May 14 20:49:26 2015 +0800
FM: respect hw.fm.internal_antenna

PS2: Patch from here: http://forum.xda-developers.com/showpost.php?p=60580714&postcount=86

Change-Id: Ic3104de252bac108c925e7cd5f50d9e89ca63779
---
 fmapp2/src/com/caf/fmradio/FMRadioService.java |  8 +++-----
 qcom/fmradio/FmTransceiver.java                | 11 ++++-------
 2 files changed, 7 insertions(+), 12 deletions(-)

diff --git a/fmapp2/src/com/caf/fmradio/FMRadioService.java b/fmapp2/src/com/caf/fmradio/FMRadioService.java
index 78bf9ab..d2d833a 100644
--- a/fmapp2/src/com/caf/fmradio/FMRadioService.java
+++ b/fmapp2/src/com/caf/fmradio/FMRadioService.java
@@ -2400,13 +2400,11 @@ public class FMRadioService extends Service
             bStatus = enableAutoAF(FmSharedPreferences.getAutoAFSwitch());
             Log.d(LOGTAG, "enableAutoAF done, Status :" +  bStatus);
 
-            /* There is no internal Antenna*/
-            bStatus = mReceiver.setInternalAntenna(false);
-            Log.d(LOGTAG, "setInternalAntenna done, Status :" +  bStatus);
-
-            /* Read back to verify the internal Antenna mode*/
             readInternalAntennaAvailable();
 
+            bStatus = mReceiver.setInternalAntenna(mInternalAntennaAvailable);
+            Log.d(LOGTAG, "setInternalAntenna done, Status :" +  bStatus);
+
             startNotification();
             bStatus = true;
          }
diff --git a/qcom/fmradio/FmTransceiver.java b/qcom/fmradio/FmTransceiver.java
index b90f033..5256060 100644
--- a/qcom/fmradio/FmTransceiver.java
+++ b/qcom/fmradio/FmTransceiver.java
@@ -28,6 +28,7 @@
 
 
 package qcom.fmradio;
+import android.os.SystemProperties;
 import android.util.Log;
 import java.io.File;
 
@@ -571,13 +572,9 @@ public class FmTransceiver
    */
    public boolean getInternalAntenna()
    {
-
-       int re = FmReceiverJNI.getControlNative (sFd, V4L2_CID_PRIVATE_TAVARUA_ANTENNA);
-
-       if (re == 1)
-         return true;
-
-       return false;
+       return ((FmReceiverJNI.getControlNative(sFd,
+                       V4L2_CID_PRIVATE_TAVARUA_ANTENNA) == 1) ||
+               SystemProperties.getBoolean("hw.fm.internal_antenna", false));
    }
 
    /*==============================================================
-- 
2.17.1

